### PR TITLE
TLSv1_2-rejected-without-TLSv1_2: stricter check

### PR DESCRIPTION
**Pull request by tomato42**
### Description

* make the script checks more strict
* test more ciphers
* make script more user-friendly (make ^C actually kill it)

### Motivation and Context

better test coverage for TLS 1.1 and TLS 1.2 servers

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [x] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [ ] OpenSSL
  - [ ] NSS
  - [x] GnuTLS
- [x] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/548)

### CODE REVIEW

File: scripts/test-TLSv1_2-rejected-without-TLSv1_2.py [hunk 1]
```diff
@@ -96,6 +99,35 @@ def main():
 
     conversations["sanity"] = conversation
 
+    # sanity check - TLS 1.1 enabled
+    conversation = Connect(host, port)
+    node = conversation
+    ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
```
Comment (nmav) [inline scripts/test-TLSv1_2-rejected-without-TLSv1_2.py]: A previous patch in this set uses CipherSuite.tls12Suites. Is there a reason restricting this sanity to a single ciphersuite?

Comment (tomato42) [reply scripts/test-TLSv1_2-rejected-without-TLSv1_2.py]: 1. it's the same cipher that is used in the `sanity` case
2. at the very least, it has to be limited to RSA key exchange ciphers as the test doesn't expect ServerKeyExchange message

so it's easier to have just one cipher here

**Comment by tomato42** [comment]
regarding stricter check: commit message updated

File: scripts/test-TLSv1_2-rejected-without-TLSv1_2.py (hunk 1 above)
Comment (The-Mule) [reply scripts/test-TLSv1_2-rejected-without-TLSv1_2.py]: Just for the record, RSA key exchange is not allowed in FIPS mode on RHEL-8.

Comment (tomato42) [reply scripts/test-TLSv1_2-rejected-without-TLSv1_2.py]: that's something we will have to address for all scripts

**Pull request merged**